drivers/gpu/drm/bridge/cdns-mhdp.c

```diff
@@ -534,8 +535,15 @@ int cdns_mhdp_adjust_lt(struct cdns_mhdp_device *mhdp,
 
 #define CDNS_KEEP_ALIVE_TIMEOUT			2000
 
+static const struct mhdp_ops mhdp_ti_ops = {
+	.init = cdns_mhdp_j721e_init,
+	.exit = cdns_mhdp_j721e_fini,
+	.enable = cdns_mhdp_j721e_enable,
+	.disable = cdns_mhdp_j721e_disable,
+};
+
 static const struct of_device_id mhdp_ids[] = {
-	{ .compatible = "cdns,mhdp8546", },
+	{ .compatible = "cdns,mhdp8546", .data = (void *)&mhdp_ti_ops },
```

Comment:
Is the (void*) needed?

I think we need an additional compatible string here for J7. I'm not sure what the exact string should be, but I think this can be handled with something like this:

"cdns,mhdp8546" means the base IP, without platform specific ops. "ti,j721e-mhdp8546" means J7's integration.

So in the above table, we can have entries for both. Only the j7 one would have the .data field filled.

In the j7 dtsi file, we can use:

compatible = "ti,j721e-mhdp8546", "cdns,mhdp8546";

But as I said, I'm not sure about the exact naming. Need to see what is used for other J7 components, I guess they may have similar problems.

Reply:
The j721e specific compatible string and the associated ops should be enclosed within #ifdef CONFIG_DRM_CDNS_MHDP_J721E.